### PR TITLE
Better setup

### PR DESCRIPTION
**Pull request by tian2992**
Please add comments & code as needed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,28 +2,34 @@
...
 Example of creating an ocdsdata table and data:
 ```
 sudo -u postgres createuser ocdsdata --pwprompt
-sudo -u postgres createdb ocdsdata -O ocdsdata --encoding UTF8 --template template0 --lc-collate en_US.UTF-8 --lc-ctype en_US.UTF-8 
+sudo -u postgres createdb ocdsdata -O ocdsdata --encoding UTF8 --template template0 --lc-collate en_US.UTF-8 --lc-ctype en_US.UTF-8
 export DB_URI='postgres://ocdsdata:PASSWORD YOU CHOSE@localhost/ocdsdata'
 python ocdsdata-cli upgrade-database
 ```
 
+<<<<<<< HEAD
```
Comment (kindly) [inline README.md]: These merge conflict marks should be removed

File: MANIFEST [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# file GENERATED by distutils, do NOT edit
```
Comment (kindly) [inline MANIFEST]: I do not think the MANIFEST should be added to github. It should be in the .gitignore

File: README.md (hunk 1 above)
Comment (tian2992) [reply README.md]: Gee! sorry for the slip.

File: MANIFEST (hunk 2 above)
Comment (tian2992) [reply MANIFEST]: I have heard diverse opinions, but I shall remove it.

File: README.md [hunk 3]
```diff
@@ -37,13 +41,16 @@ The order of precedence is (from least-important to most-important):
   *  password from ~/.pgpass
   *  environmental variable
 
-## Status of a run
-
-During or after a run you can use a command to check on the progress.
```
Comment (odscjames) [inline README.md]: This line should stay

File: README.md [hunk 4]
```diff
@@ -37,13 +41,16 @@ The order of precedence is (from least-important to most-important):
...
+Run `ocdsdata-cli` with the argument of one of the publishers you want to fetch.
 
 Run `ocdsdata-cli status` with the source flag as the publisher you want to see. Pass the sample flag too, if it's a sample run.
 
-By default it will show the progress for the latest run, but you can pass the dataversion flag to see different ones.
```
Comment (odscjames) [inline README.md]: This line shouldn't go

File: README.md [hunk 5]
```diff
@@ -2,28 +2,32 @@
...
 Example of creating an ocdsdata table and data:
 ```
 sudo -u postgres createuser ocdsdata --pwprompt
-sudo -u postgres createdb ocdsdata -O ocdsdata --encoding UTF8 --template template0 --lc-collate en_US.UTF-8 --lc-ctype en_US.UTF-8 
+sudo -u postgres createdb ocdsdata -O ocdsdata --encoding UTF8 --template template0 --lc-collate en_US.UTF-8 --lc-ctype en_US.UTF-8
 export DB_URI='postgres://ocdsdata:PASSWORD YOU CHOSE@localhost/ocdsdata'
 python ocdsdata-cli upgrade-database
 ```
 
+=======
```
Comment (odscjames) [inline README.md]: This is an artifact of the Git conflict I think? I don't think this should be here

File: README.md [hunk 6]
```diff
@@ -37,13 +41,16 @@ The order of precedence is (from least-important to most-important):
...
-
-During or after a run you can use a command to check on the progress.
+Run `ocdsdata-cli` with the argument of one of the publishers you want to fetch.
 
 Run `ocdsdata-cli status` with the source flag as the publisher you want to see. Pass the sample flag too, if it's a sample run.
 
-By default it will show the progress for the latest run, but you can pass the dataversion flag to see different ones.
+## Configuration
+
+Database and configuration settings can be set using `~/.config/ocdsdata/config.ini` file. A sample one is included in the main directory.
```
Comment (odscjames) [inline README.md]: This section is a repeat of something above - I guess Git merge issues?

**Comment by odscjames** [comment]
Unfortunately there are still problems with the readme - some sections are still repeated. Sorry!

Can I suggest just copying and pasting https://github.com/open-contracting/ocdsdata/blob/master/README.md to https://github.com/open-contracting/ocdsdata/blob/better-setup/README.md ?

I  think only one thing in the README.md actually needs to change as a result of this pull request.

I see you added "pip install -e ." to the first section, which I guess is needed because "requirements.txt" has had  "--e ." removed? 

So copy and pasting and then changing that one thing may be the easiest way to solve this merge. Thank you :-)

**Comment by odscjames** [comment]
Firstly Sebastian, sorry this has taken so long. Honestly, I'm not an expert in Python packaging so my reviews of this have been slower than I would have liked.

(I pushed a commit to fix the README.md - the new pip line needed wasn't in, and some of the other lines had changed)

I have been asking others in an attempt to get help, and from that I have some questions:

1) it's been pointed out that the setup.py says:

> Note: this will only work if 'README.md' is present in your MANIFEST.in file!

However, there is no MANIFEST.in file in this pull request? Should there be?

2) We weren't sure if the contents of requirements.txt should be in setup.py, in the install_requires key?

3) We weren't sure if the "-e ." should be taken out of the requirements.txt - as that then needs extra instructions added to the readme file. Is there a good reason to do that? Or should the requirements.txt have just "-e ." and then leave the setup.py to install everything else?

4) Can we take both ''Intended Audience' classifications out of setup.py? Neither of them are quite right - looking at the list it's not clear there is a perfect one, so we will just leave it for now.

Again, sorry for the delay! With some discussion we can sort this soon.

**Comment by tian2992** [comment]
1. The Manifest was removed at @kindly 's request. I think it is a good idea to have one, which at least includes a list of routes, but it is fine either way.. https://github.com/open-contracting/ocdsdata/pull/132#discussion_r199165246

2. &amp;  3. The rationale for leaving out the `-e .` is: `-e.` should only be run when installing in development mode, making it explicit as a command is good. When we have a deployment cycle, we want most people to be installing our tool from packaged sources (either PIP or a dist..?) When running `pip install ocdsdata` e.g. The other reason was to be able to parse REQUIREMENTS and load them into setup.py at "runtime", this means less problem when changing/updating deps, this is still lacking code, however... I am in favour of using install_requires, as it allows us to keep both the convenience of requirements.txt as well as ensuring both install methods are supported.

&nbsp;4. Surely, will push ASAP.

What do you think could be a good course of action.

**Comment by tian2992** [comment]
@odscjames i dunno how to re-ask you for review via the UI, sorry if dismissing the review seems rude .

**Comment by odscjames** [comment]
> The Manifest was removed at @kindly 's request.

He said to remove the MANIFEST, this comment refers to MANIFEST.in - different files.

However, I don't think we need one. https://packaging.python.org/guides/distributing-packages-using-setuptools/#readme-rst-readme-md says that README's are included automatically in later versions. (This explains @jpmckinney comment about ocdskit) So in fact, can we remove the comment:

 > Note: this will only work if 'README.md' is present in your MANIFEST.in file!

As that appears not to be true any more.

So that's one change.

Other steps:

> ocdskit's .travis.yml does pip install without -e (no need for editable mode for automated testing).

Yup, lets do that as a second change.

> However, flatten-tool and CoVE should ideally make releases

I think we should aim to get this P.R. in without this to start with, as this won't be happening shortly. We have started on this for flatten-tool but it will take longer to get CoVE sorted out for various reasons. (Actually, I'd prefer to solve this problem a different way but that's for a seperate issue.)

> I agree that requirements.txt should only contain any requirements that cannot be in setup.py 
> Non-essential 'requirements' like flake8 should be in extra_requires.

Given the last point, and because I'm aware this has gone on a bit (Sorry!) let's leave this for now and come back to this when we can achieve more by this move? Is that ok?

So with those 2 changes, shall we merge this in?

Hint: If #175 is merged to master it will make another merge conflict here - so if it is possible to make these 2 changes quickly I'll merge this in first so that I have to take the hit of fixing the conflict in #175 instead!

**Pull request merged**